src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,282 @@
+import { Box, ChevronIcon, Flex, space } from "@artsy/palette"
+import { Options as FlickityOptions } from "flickity"
+import React, { ReactNode, useRef, useState } from "react"
+import styled from "styled-components"
+import { left, LeftProps, right, RightProps } from "styled-system"
+import { useDidMount } from "Utils/hooks/useDidMount"
+import { Media } from "Utils/Responsive"
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  options?: FlickityOptions
+}
+
+const isClient = typeof window !== "undefined"
+
+/*
+ * Flickity docs https://flickity.metafizzy.co
+ * Flickity requires window, so we must render a seperate Flex wrapper instead of
+ * a carousel if we're using SSR.
+ */
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+  state = {
+    isMounted: false,
+  }
+
+  componentDidMount() {
+    this.setState({
+      isMounted: true,
+    })
+  }
+
+  /*
+   * If we don't check for isMounted, React will not properly clean up the
+   * CarouselServer component on initial render
+   */
+  render() {
+    return (
+      <>
+        {isClient && this.state.isMounted && (
+          <Box key="client">
+            <Media at="xs">
+              <SmallCarousel {...this.props} />
+            </Media>
+            <Media greaterThan="xs">
+              <LargeCarousel {...this.props} />
+            </Media>
+          </Box>
+        )}
+        {!isClient && !this.state.isMounted && (
+          <Box key="server">
+            <CarouselServer {...this.props} />
+          </Box>
+        )}
+      </>
+    )
+  }
+}
+
+const renderSlides = props => {
+  return props.data.map((slide, index) => {
+    return <Box key={index}>{props.render(slide)}</Box>
+  })
+}
+
+/*
+ * This version of the carousel is only rendered for SSR
+ */
+const CarouselServer = (props: Props) => {
+  return (
+    <Flex
+      flexDirection="row"
+      position="relative"
+      justifyContent="space-around"
+      alignItems="center"
+      height={props.height}
+    >
+      <CarouselContainer height={props.height}>
+        <Flex>{renderSlides(props)}</Flex>
+      </CarouselContainer>
+    </Flex>
+  )
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity = isClient ? require("react-flickity-component") : () => null
+
+export const LargeCarousel = (props: Props) => {
+  const [currentSlideIndex, setSlideIndex] = useState(0)
+  const [lastItemVisible, setLastItemVisible] = useState(false)
+  const flicktyRef = useRef(null)
+  const isMounted = useDidMount()
```

Comment:
Rather than using this `isMounted` can access via props passed down from the component's state up above. Can also delete this `useDidMount` file since we're no longer using it.